src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+});
+
+profileAddButton.addEventListener("click", () => {
+  openModal(newPostModal);
+});
+
+profileExitButton.addEventListener("click", () => {
+  closeModal(profileModal);
+});
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+deleteExitBtn.addEventListener("click", () => {
+  closeModal(deleteCardModal);
+});
+
+deleteCancel.addEventListener("click", () => {
+  closeModal(deleteCardModal);
+});
+
+function handleAvatarSubmit(evt) {
+  // form for changing Avatar
+  evt.preventDefault();
+  setButtonText(avatarSubmitButton, true);
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      avatarImg.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitButton, false);
+    });
+}
+
+function handleNewPost(evt) {
+  //new post modal submit
+  evt.preventDefault();
+  setButtonText(cardSubmitButton, true);
+
+  const name = caption.value;
+  const link = imageLink.value;
+  api
+    .createNewCard(name, link)
+    .then((card) => {
+      const cardsElement = getCardElement(card);
+
+      cardsList.prepend(cardsElement);
+      evt.target.reset();
+      closeModal(newPostModal);
+      resetValidation(evt.target, validationConfig);
+      disableButton(cardSubmitButton);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitButton, false);
+    });
+}
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+modalImageForm.addEventListener("submit", handleNewPost);
+
+profileEditButton.addEventListener("click", () => {
+  profileName.value = nameOnPage.textContent;
+  profileDescription.value = descriptionOnPage.textContent;
+
+  openModal(profileModal);
+  resetValidation(profileModal, validationConfig);
+});
+
+function handleProfileFormSubmit(evt) {
+  //form for profile name and description
+  evt.preventDefault();
+  setButtonText(profileSubmitBtn, true);
+
+  api
+    .editUserInfo({ name: profileDescription.value, about: profileName.value })
+    .then((data) => {
+      nameOnPage.textContent = data.about;
+      descriptionOnPage.textContent = data.name;
+      closeModal(profileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(profileSubmitBtn, false);
+    });
+}
+
+profileForm.addEventListener("submit", handleProfileFormSubmit);
+
+function handleDeleteCard(cardElement, cardId) {
+  // form for deleting cards
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteCardModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  setSubmitText(deleteSubmitBtn, true);
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setSubmitText(deleteSubmitBtn, false);
```

Comment:
`setSubmitText` is not needed because it duplicates the same `setButtonText` function. To make it work for the deleting modal, you need to pass the 3rd and 4th arguments


```js
    .finally(() => {
      setButtonText(deleteSubmitBtn, false, "DELETE", "DELETING...");
    });

```